src/broker/broker.c

```diff
@@ -834,6 +839,73 @@ static int init_local_uri_attr (struct overlay *ov, attr_t *attrs)
     return 0;
 }
 
+static int send_uri_job_memo (attr_t *attrs)
+{
+    const char *jobid = NULL;
+    const char *parent_uri = NULL;
+    const char *local_uri = NULL;
+    const char *path;
+    char uri [1024];
+    char hostname [MAXHOSTNAMELEN + 1];
+    flux_jobid_t id;
+    flux_t *h;
+    flux_future_t *f;
+
+    /* Skip if "jobid" or "parent-uri" not set, this is probably
+     *  not a child of any Flux instance.
+     */
+    if (attr_get (attrs, "parent-uri", &parent_uri, NULL) < 0
+        || parent_uri == NULL
+        || attr_get (attrs, "jobid", &jobid, NULL) < 0
+        || jobid == NULL)
+        return 0;
+
+    if (flux_job_id_parse (jobid, &id) < 0) {
+        log_err ("Unable to parse jobid attribute '%s'", jobid);
+        return -1;
+    }
+    if (attr_get (attrs, "local-uri", &local_uri, NULL) < 0) {
+        log_err ("Unexpectedly unable to fetch local-uri attribute");
+        return -1;
+    }
+    if (gethostname (hostname, sizeof (hostname)) < 0) {
+        log_err ("gethostname failure");
+        return -1;
+    }
+    path = local_uri + 8; /* forward past "local://" */
+    if (snprintf (uri,
+                 sizeof (uri),
+                 "ssh://%s%s",
+                 hostname, path) >= sizeof (uri)) {
+        log_msg ("buffer overflow while checking local-uri");
+        return -1;
+    }
+
+    /*  Open connection to parent instance and post "uri" user annotation
+     */
+    if (!(h = flux_open (parent_uri, 0))) {
+        log_err ("flux_open to parent failed");
+        return -1;
+    }
+    if (!(f = flux_rpc_pack (h,
+                             "job-manager.memo",
+                             FLUX_NODEID_ANY,
+                             0,
+                             "{s:I s:{s:s}}",
+                             "id", id,
+                             "memo",
+                               "uri", uri)))
+        return -1;
```

Comment:
need to call `flux_close()` on `h` before returning

Reply:
Got it, good catch.